packages/@aws-cdk/aws-appmesh/lib/virtual-node.ts

```diff
@@ -33,10 +33,449 @@ export interface IVirtualNode extends cdk.IResource {
    */
   addBackends(...props: IVirtualService[]): void;
 
+  /**
+   * Utility method to add a single node listener
+   */
+  addListener(listener: VirtualNodeListener): void;
+
   /**
    * Utility method to add Node Listeners for new or existing VirtualNodes
    */
-  addListeners(...listeners: VirtualNodeListener[]): void;
```

Comment:
I still haven't seen a reply to this 🙂

Reply:
Without rest parameter: `addListener(listeners: VirtualNodeListener[])`
```
node.addListeners([appmesh.VirtualNodeListener.tcp({
  port: 80,
  healthCheck: { timeout: cdk.Duration.seconds(3) },
  timeout: { idle: cdk.Duration.seconds(10) },
}),
appmesh.VirtualNodeListener.http2({
  port: 80,
  healthCheck: {},
  timeout: { idle: cdk.Duration.seconds(10) },
})]);
```
With rest parameter: `addListeners(...listener: VirtualNodeListener[])`

```
node.addListeners(appmesh.VirtualNodeListener.tcp({
  port: 80,
  healthCheck: { timeout: cdk.Duration.seconds(3) },
  timeout: { idle: cdk.Duration.seconds(10) },
}),
appmesh.VirtualNodeListener.http2({
  port: 80,
  healthCheck: {},
  timeout: { idle: cdk.Duration.seconds(10) },
}));
```
If I understand correctly, there are not differences in using either one of them except one would accept array of VirtualListenerNode and other would accept any number of VirtualNodeListener type objects. But to introduce less changes to the existing code, I guess we should use the rest parameters or is this change acceptable ?

Reply:
My question is, should we be adding this method to the `IVirtualNode` class at all? It would not make sense for an imported virtual node to add listeners, or even backends since those would need to be able to mutate the underlying resource.

Reply:
> My question is, should we be adding this method to the `IVirtualNode` class at all? It would not make sense for an imported virtual node to add listeners, or even backends since those would need to be able to mutate the underlying resource.

In that case, we should move this method to the `VirtualNode` class, instead of having it on `IVirtualNode`.

Can Virtual Nodes today have more than one Listener?

Reply:
Not today, but similar to other resources we have supporting multiple listeners on the roadmap

Reply:
In that case, let's remove the `addListeners` method completely, and move the `addListener` from `IVirtualNode` to `VirtualNode`.

Reply:
We need a breaking change in the PR description for this as well.